src/jit/emitxarch.cpp

```diff
@@ -2975,7 +2975,6 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
                 // src is a class static variable
                 // dst is a register
                 emitIns_R_C(ins, attr, dst->gtRegNum, memBase->gtClsVar.gtClsVarHnd, 0);
-                codeGen->genProduceReg(dst);
```

Comment:
Here's what I put in the commit message:

```
This issue did not appear on x64 because
AFAICT we very nearly always morph class field loads into loads of
a `CNS_INT` handle (see morph.cpp:6344 in `fgMorphField` for
details).
```